### DISCUSSION TITLE
Passing workers to Axum handlers

### DISCUSSION THREAD

**Discussion by kdumontnu** [rscarson/rustyscript#227] [Q&A]
I had the idea to try and create an SSR Vue web app with Axum, using this library as a JS runtime to build & hydrate the Vue code [here](https://vuejs.org/guide/scaling-up/ssr.html). 

I think this could be a really neat use-case for this library, but I haven't been able to successfully pass workers to Axum handlers threads. I'm trying to wrap my head around what the appropriate implementation would be wrt rustyscript.

I tried putting the code from [default threaded workers](https://github.com/rscarson/rustyscript/blob/master/examples/default_threaded_worker.rs) example in an axum handler, and it actually seems to run the first time the route is called, but the second time I call it, it seg-faults. 

I get the same result if I initialize two workers on the same thread. Opened an [issue here](https://github.com/rscarson/rustyscript/issues/228).

Trying to see if anyone has seen something similar or has ideas moving forward.

**Comment by rscarson** [comment]
It is worth noting that workers are always on separate threads

Can you post the code for the handler that crashes?
